### PR TITLE
safety check for null values

### PR DESCRIPTION
**Pull request by cflynn07**

### CODE REVIEW

File: test/flatten.js [hunk 1]
```diff
@@ -50,6 +50,31 @@ describe('flatten', function () {
         });
       });
     });
+    describe('null', function () {
+      before(function () {
+        this.obj = {
+          num: new Number(1), // ignore
```
Comment (bkendall) [inline test/flatten.js]: why is this ignored by keypather? I don't see that anywhere else, so I'm curious... I see `wiz: null` carries through...

File: index.js [hunk 2]
```diff
@@ -91,7 +91,7 @@ Keypather.prototype.flatten = function (obj, delimeter, preKeypath, init) {
     var keypath = exists(preKeypath) ?
       [ preKeypath, key ].join(arr ? '' : delimeter) :
       key;
-    if (typeof val === 'object') {
+    if (val && typeof val === 'object') {
```
Comment (tjmehta) [inline index.js]: Maybe use `101/is-object` here. It is the most strict object check.

File: test/flatten.js (hunk 1 above)
Comment (tjmehta) [inline test/flatten.js]: Note the comment above

**Comment by tjmehta** [comment]
merge conflict

File: test/flatten.js (hunk 1 above)
Comment (bkendall) [inline test/flatten.js]: well, I guess that `ignore` is an... interesting comment. it doesn't ignore it, it just uses it straight up (like the value at `foo.qux`). If I was looking at the tests for an example and saw this comment, I may come away confused (as I have)

Comment (tjmehta) [inline test/flatten.js]: The comment is out of date



—@tjmehta
Typed using my thumbs..

**Comment by cflynn07** [comment]
Ping, 7 months later. Could use keypather.flatten with what I'm currently working on.

**Comment by tjmehta** [comment]
At quick glance this looks good. I am out of town and flying tomorrow. Please bump if I forget

**Comment by tjmehta** [comment]
:+1:

**Comment by tjmehta** [comment]
keypather@v1.10.2

**Pull request merged**